resources/schema.py

```diff
@@ -328,6 +348,140 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return DeleteBerthTypeMutation()
 
 
+class CreateHarborMutation(graphene.ClientIDMutation):
+    class Input:
+        area = HarborAreaInput(required=True)
+        municipality = graphene.String()
+        image_file = graphene.String()
+        availability_level = graphene.ID()
+        number_of_places = graphene.Int()
+        maximum_width = graphene.Int()
+        maximum_length = graphene.Int()
+        maximum_depth = graphene.Int()
+        name = graphene.String()
+        street_address = graphene.String()
+
+    harbor = graphene.Field(HarborNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        lang = info.context.META.get("HTTP_ACCEPT_LANG", settings.LANGUAGE_CODE).split(
+            "-"
+        )[0]
+
+        area = kwargs.pop("area", {})
+
+        try:
+            availability_level_id = kwargs.pop("availability_level", None)
```

Comment:
ideally, this would be outside of the `try` block:
```python
availability_level_id = kwargs.pop("availability_level", None)
if availability_level_id:
    try:
        availability_level = AvailabilityLevel.objects.language(lang).get(
                    pk=availability_level_id
                )
        except AvailabilityLevel.DoesNotExist as e:
            raise VenepaikkaGraphQLError(e)
```
this way you might save avoid starting the routine, when it's not needed 🙂